### ISSUE TITLE
Cross entropy loss function

### ISSUE THREAD

**Issue by tmeloliveira** [makeyourownneuralnetwork/makeyourownneuralnetwork#13]
Hello! 

First of all, congratulations on writing such a great book! It helped me a lot to understand Neural Networks. For me that's exactly how learning should be! :) 

Second, I have a quick question... Is it possible to adapt code for the neural network to use cross-entropy as a loss function? Would it change much in the code?

As I am still studying, I have read in some places that cross-entropy function works very well... So I would like to compare it with the performance of the one we used in the book. 

Thanks in advance!

**Comment by makeyourownneuralnetwork** [comment]
Hi Thiago

I'm glad the book worked for you ... it was my intention to help readers
get onto the first steps of this exciting world!

Yes - the cross entropy function can be used and is better for some
problems. That's the key - you have to know when and why the cross entropy
function is better for your problem.

The changes are easily made :

1. The error function is now derived from the cross entropy function.

2. The weight update formula must then also change to match the new error
function.

Some people only apply the cross entropy at the last layer .. in which case
you'll need to keep the traditional error function inside the network and
the weight update formula .. based on the logistic or tanh or other
activation function.

You'll find loads of worked examples on the web.

I hope this helps!

Tariq

**Comment by anthony123** [comment]
As you can see in the book , the part derivative expression composed of three part, firstly,  the error part: (target-actual), secondly, the sigmoid part: sigmoid(z)(1-sigmoid(z)), and thirdly the output of previous layer O(i) 

when we use the cross entropy loss, the second part can be cancelled out. left only with the first and the last part, which is good. Because the revised expression is not depend on the sigmoid function, making the learning faster.
